src/rdo.rs

```diff
@@ -585,6 +585,15 @@ impl DistortionScale {
     )
   }
 
+  pub fn inv_mean(slice: &[Self]) -> Self {
+    let sum: u64 = slice.iter().map(|&s| s.0 as u64).sum();
+    assert_ne!(sum, 0);
```

Comment:
I wonder if there's any benefit from using `NonZeroU64` here. Maybe it makes the code look cleaner? I'm not sure if the compiler benefits from it or if it considers `assert_ne!(sum, 0)` equivalent for its optimization routine.

Reply:
I think there is an argument for changing the inner type of `DistortionScale` to `NonZeroU32`.

Reply:
I opted to simply ensure that the outputs of the new operations are always in range.